homeassistant/components/advantage_air/climate.py

```diff
@@ -110,32 +113,32 @@ class AdvantageAirAC(AdvantageAirClimateEntity):
     _attr_hvac_modes = AC_HVAC_MODES
     _attr_supported_features = SUPPORT_TARGET_TEMPERATURE | SUPPORT_FAN_MODE
 
-    def __init__(self, instance, ac_key):
+    def __init__(self, instance: advantage_air, ac_key: int) -> None:
         """Initialize an AdvantageAir AC unit."""
         super().__init__(instance, ac_key)
-        self._attr_name = self._ac["name"]
+        self._attr_name = cast(str, self._ac["name"])
         self._attr_unique_id = f'{self.coordinator.data["system"]["rid"]}-{ac_key}'
         if self._ac.get("myAutoModeEnabled"):
             self._attr_hvac_modes = AC_HVAC_MODES + [HVAC_MODE_AUTO]
 
     @property
-    def target_temperature(self):
+    def target_temperature(self) -> int:
```

Comment:
Is it an integer though? The set target temp seems to handle a float?

Reply:
So I certainly know the product API returns either an integer or float depending on which model it is, but as far as ive seen it will always return whole numbers, which is why I use `_attr_target_temperature_step = PRECISION_WHOLE`.